src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp

```diff
@@ -179,8 +179,23 @@ void AwtWin32GraphicsDevice::Initialize()
     }
     gpBitmapInfo->bmiHeader.biBitCount = 0;
     HDC hBMDC = this->GetDC();
+    VERIFY(hBMDC != NULL);
+    if (hBMDC == NULL) {
+        return;
+    }
     HBITMAP hBM = ::CreateCompatibleBitmap(hBMDC, 1, 1);
-    VERIFY(::GetDIBits(hBMDC, hBM, 0, 1, NULL, gpBitmapInfo, DIB_RGB_COLORS));
+    VERIFY(hBM != NULL);
+    if (hBM == NULL) {
+        VERIFY(::DeleteDC(hBMDC));
+        return;
+    }
+    int getDiBitsRC = ::GetDIBits(hBMDC, hBM, 0, 1, NULL, gpBitmapInfo, DIB_RGB_COLORS);
+    VERIFY(getDiBitsRC);
+    if (getDiBitsRC == 0) {
+        VERIFY(::DeleteObject(hBM));
+        VERIFY(::DeleteDC(hBMDC));
+        return;
+    }
 
     if (colorData->bitsperpixel > 8) {
```

Comment:
So at this time, the code below will not be run if an error occurs. This may lead to leaving the fields of `gpBitmapInfo` uninitialised.

I'm unsure how critical it is… even though we're (presumably) on a headless system.

Reply:
OK, yes, I'll turn this back to previous behavior. Rather don't touch something at all if we have no idea... 😉

Reply:
I prefer *backing out*… — it makes the code cleaner. At the same time, I am unsure about the consequences of doing so. I could be safer to run the code below, especially if you take into account that there's a fallback already:

https://github.com/openjdk/jdk/blob/fb90af881badf143163c7d0b9961152c2a12cd84/src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp#L192-L199

Reply:
Reverted that stuff.